src/mpid/ch4/netmod/ofi/ofi_am_events.h

```diff
@@ -342,7 +342,7 @@ static inline int MPIDI_OFI_handle_lmt_ack(MPIDI_OFI_am_header_t * msg_hdr)
         MPIR_Assert(MPIDI_OFI_KEY_TYPE_HUGE_RMA == key_type);
 
         MPIDI_OFI_index_allocator_free(MPIDI_OFI_COMM(MPIR_Process.comm_world).rma_id_allocator,
-                                       idx_back);
+                                       (uint64_t) idx_back);
```

Comment:
Why are we casting here? Are we always truncating the allocated key and casting it back? That seems weird.

Reply:
Hmm, I think I need to take a another look of the implementation.

Reply:
The casting here is just suppressing the warning. Even without the casting, the compiler would do an implicit casting.